src/app/shared/components/template/components/data-items/data-items.utils.ts

```diff
@@ -69,7 +69,7 @@ export function updateItemMeta(
     }
 
     // Apply recursively to ensure item children with nested rows (e.g. display groups) also inherit item context
-    if (r.rows) {
+    if (r.rows && r.type !== "data_items" && r.type !== "items") {
```

Comment:
Might benefit from an explanatory comment on why `data_items` and `items` rows are excluded

Reply:
Updated in https://github.com/IDEMSInternational/open-app-builder/pull/2647